src/services/firebase.ts

```diff
@@ -112,7 +118,54 @@ export namespace ourFirebase {
     // Make sure we listen to match changes only once.
     // 'gamePortal/matches'
     // store.dispatch(updateMatchList);
-    db().ref('gamePortal/matches' + matchIds); // TODO
+    let tempMatchIds: Promise<MatchInfo>[] = [];
+    for (let matchId of matchIds) {
+      tempMatchIds.push(getMatchDetail(matchId));
+    }
+    Promise.all(tempMatchIds).then((matches: MatchInfo[]) => {
+      let action: Action = {
+        setMatchesList: matches
+      };
+      dispatch(action);
+    });
+
+    // db().ref('gamePortal/matches' + matchIds); // TODO
+  }
+
+  function getMatchDetail(matchId: string): Promise<MatchInfo> {
+    // let matchInfo = {};
+    return db()
+      .ref('gamePortal/matches/' + matchId)
+      .once('value')
+      .then((snap: firebase.database.DataSnapshot): MatchInfo => {
+        let matchFb: fbr.Match = snap.val();
+        let gameSpecId = matchFb.gameSpecId;
+        let gameSet = store.getState().gamesList;
+        let game = gameSet[gameSpecId];
+        let newMatchStates: MatchState = {};
+        let tempPieces = matchFb.pieces;
+        for (let tempPieceKey of Object.keys(tempPieces)) {
+          let newMatchState: PieceState;
```

Comment:
directly create.

Reply:
Hi Amanpreet,

Do you have time today? Can I ask you some questions about test?

Reply:
You should email me and professor. I am stuck with work of my own, so won't be coming to New York today.

Reply:
For testing this, you will need to use `done` as a parameter to your test. Which is function to be called once your async calls are complete and all of testing it done. See the documentation of jest. It is pretty well explained.

Reply:
Okay. I have written a test function but it always fail. I guess it may be the reason of the functions I wrote in the firebase.ts. I'm checking it right now.